Orange-OpenSource/Orange-Boosted-Bootstrap
# 🚀 v5.2.1

## Specific to this version

- [x] The content of this v5.2.1 is defined and definitive
- [x] Merge the remaining PRs
  - [x] #1579 
  - [x] #1088 
  - [x] #1522
- [ ] Following `gh-pages` modifications have been done before merging this PR:
  - [ ] https://github.com/twbs/bootstrap/commit/5cff761c91cea600f410d88a3eff32241d2d3bcd
  - [ ] https://github.com/twbs/bootstrap/commit/c2b4f59e0c0e6de08510d51ef37d04cca41c22c6
- [ ] When merging to `gh-pages` check the diff with https://github.com/twbs/bootstrap/commit/1510239c8f6406ce07f50f5fbc1242767c07d397 

## Generic

  - Firefox ESR
  - IE11 (v4 only)
  - Latest Edge, Chrome, Firefox, Safari
  - iOS Safari
  - Chrome & Firefox on Android
  - then, if bumping a minor or major version:
    - ~~Manually change `version_short` in `package.json`~~
    - ~~Add docs version to `site/data/docs-versions.yml`~~
    - ~~Manually change `docs_version` in `config.yml` and other references to the previous version~~
    - ~~Update redirects in docs frontmatter (`site/content/docs/_index.html`?)~~
    - ~~Move `site/content/docs/5.x` to `site/content/docs/5.x+1`~~
    - ~~Increment `site/static/docs/{version}` version~~
    - ~~Increment version in `nuget/boosted.nuspec`~~
    - ~~Ensure Algolia indexes new release content ([probably requires a PR](https://github.com/algolia/docsearch-configs/pull/1954))~~
    - ~~(Major version) Manually update the version in `nuget/boosted.nuspec` and `nuget/boosted.sass.nuspec`~~
  - check wrong matches in `CHANGELOG.md`, and maybe `site/content/docs/<version>/migration.md`
  - :warning: check the `package-lock.json` and `package.json` content, only "boosted" should have its version changed!
  - :warning: `site/content/docs/5.1/**/*.md` should not always be modified
  - install [Conventionnal Changelog](https://github.com/conventional-changelog/conventional-changelog) and `conventional-changelog-cli` globally
  - run `conventional-changelog -p angular -i CHANGELOG.md -s`
  - and probably maintain [a ship list (eg for v4.4.0)](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/226)
  - attach zip file
  - paste CHANGELOG / Ship list in the release's description
  - `npm pack`
  - if you are already logged in NPM (with a personnal account, for example), [you'd better use a repository scoped `.npmrc` file](https://stackoverflow.com/questions/30114166/how-to-have-multiple-npm-users-set-up-locally)
  - Publish:
    - if you're releasing a pre-release, use `--tag`, eg for v5-alpha1 `npm publish boosted-5.0.0-alpha1.tgz --tag next`
    - (v4 only) `npm publish --tag v4.x.y` (if you forgot and v4 becomes the latest version on NPM, you can run `npm dist-tag add boosted@5.x.y latest to fix it)
    - (v5 only) `npm publish`
  - [ ] copy `../_site` to the `gh-pages` branch
